client/src/sections/resources/UndergroundFireworksShop.tsx

```diff
@@ -0,0 +1,46 @@
+import Line from "../../components/Line";
+import Tile from "../../components/Tile";
+import useHave from "../../hooks/useHave";
+import { useProperties } from "../../hooks/useProperties";
+import { $effect } from "../../util/makeValue";
+
+const UndergroundFireworksShop = () => {
+  const { _fireworksShopEquipmentBought, _fireworksShopHatBought } =
+    useProperties({
+      _fireworksShopEquipmentBought: false,
+      _fireworksShopHatBought: false,
+    });
+  const everythingLooksYellow = useHave($effect`Everything Looks Yellow`);
+  const everythingLooksBlue = useHave($effect`Everything Looks Blue`);
+  const everythingLooksRed = useHave($effect`Everything Looks Red`);
```

Comment:
Can you add a check here to return `<></>` if none of the lines are going to render?